### PR TITLE
feat(css): handle css variables syntax

### PR DESCRIPTION
**Pull request by thanoskrg**
This PR addresses the https://github.com/highlightjs/highlight.js/issues/3237 so that CSS variables syntax is properly highlighted.

### Changes
* Update `src/languages/css.js` definition
* Update test scenarios for CSS markup

### Checklist
- [x] Added markup tests
- [x] Updated the changelog at `CHANGES.md`

### CODE REVIEW

**Comment by joshgoebel** [comment]
A visual snap is alwyas nice with PRs like this.

File: src/languages/css.js
Comment (joshgoebel) [inline src/languages/css.js:30]: This looks a bit overly complex a regex... what are we trying to accomplish with the `(?!\s)`? Must variables be two letters at least?

**Comment by thanoskrg** [comment]
> A visual snap is alwyas nice with PRs like this.

@joshgoebel here it is how it looks like on *base16/nord* theme.

demo-css-variables

File: src/languages/css.js
Comment (thanoskrg) [reply src/languages/css.js:30]: > Must variables be two letters at least?

No, there is no such constraint.

I think you are right about the complexity of the regex. Maybe we could start with a simpler regex for now. 

What do you think of this?
```suggestion
    begin: /--[A-Za-z][A-Za-z0-9_-]*/,
```

Comment (joshgoebel) [reply src/languages/css.js:30]: That looks much more like what I would have expected to see.

**Comment by thanoskrg** [comment]
That's how it looks now after the latest changes:

demo-css-variables-2

File: src/languages/lib/css-shared.js
Comment (thanoskrg) [inline src/languages/lib/css-shared.js:36]: [note] @joshgoebel I thought it would make sense to place this under `lib/css-shared.js` for DRY.

**Comment by joshgoebel** [comment]
Theme maintainers: Here is a case where we're increasing the semantic density of our highlighting, but at the same time perhaps making existing theme behavior worse.  I have a suspicion many/some themes only think about `attribute` when it comes to CSS... and haven't given any thought to `attr` at all (in the context of CSS).  Given how some themes almost contain "micro-themes" for different types of markup (Markdown vs diffs vs Code).  I'm not sure of the perfect solution.

Just sticking with `attribute` here for theme behavior has the best short-term behavior but I'm not sure is a great long-term plan.  Perhaps the real problem is our historic distinction between `attr` and `attribute`... already having a clear semantic scope to reach for precludes something more complex like `attribute.custom`... thought perhaps some would argue that's better than breaking themes in the short-term?  Though it also (in the short or mid-term) makes the distinction meaningless until some future themes comes along to express the styles differently.

I could also see many themes wanting to just merge these two styles (in the case of CSS), yes I hesitate to do that at the grammar level since I always try to be as accurate as possible when it comes to scopes - and then let themes decide HOW to highlight those scope.


So perhaps the semantic difference here between attributes and custom attributes (specifically w.r.t. CSS) just doesn't matter? Thoughts?

CC @highlightjs/theme-maintainers  @highlightjs/core

**Comment by thanoskrg** [comment]
@joshgoebel any updates on this? Is this currently blocked until we get an answer on your last [question](https://github.com/highlightjs/highlight.js/pull/3239#issuecomment-860904286)?

**Comment by joshgoebel** [comment]
I was hoping for some thoughts, yes.  Eventually if there is nothing this'll probably merge as-is...  I don't usually get in a  rush until I have a new release in sight.  I need to go figure out when 11.1 will be.

**Comment by thanoskrg** [comment]
> I was hoping for some thoughts, yes. Eventually if there is nothing this'll probably merge as-is... I don't usually get in a rush until I have a new release in sight. I need to go figure out when 11.1 will be.

Thanks for your swift reply.

**Review by joshgoebel — APPROVED** [review]

**Pull request merged**
